public/firebase.js

```diff
@@ -0,0 +1,218 @@
+  const password = document.getElementById("password").value;
+  const names = document.getElementById("names").value;
+
+  createEmailPass(email, password, names);
+}
+
+//Save user by login
+function saveDataUser(user){
+  let userNew = {
+    uid:user.uid,
+    name:user.displayName,
+    photo:user.photoURL
+  }
+  db.collection("users").doc(userNew.uid).set(userNew)
+  .then(function() {
+    console.log("Document successfully written!");
+  })
+  .catch(function(error) {
+    console.error("error save user", error);
+     });
+  }
+
+  //User login e-mail
+
+  function loginUser(email, password){
+    firebase.auth().signInWithEmailAndPassword(email, password)
+    .then(function(user){
+      window.open('#/', '_self');
+    }).catch(function(error) {
+      let errorCode = error.code;
+      let errorMessage = error.message;
+    });
+  }
+
+  function loginUserEmail(){
+    const email = document.getElementById("e-mail").value;
+    const password = document.getElementById("password").value;
+  
+    loginUser(email, password);
+  }
+
+  function signOff(){
+    firebase.auth().signOut().then(function() {
+      window.open('#/login', '_self');
+    }).catch(function(error) {
+    });
+    
+  } 
+
+//add post user
+function addUserPost(){
+  let user = firebase.auth().currentUser;
+  let post = document.getElementById("addNewPost").value;
+  db.collection("post").add({
+    post: post,
+    uid: user.uid,
+    user: user.displayName,
+    imagen: user.photoURL
+})
+.then(function(docRef) {
+  console.log("Document written with ID: ", docRef.id);  
+  document.getElementById("addNewPost").value = '';
+    showPostUser();
+    
+})
+.catch(function(error) {
+    console.error("Error adding document: ", error);
+});
+}
+
+//Show post user
+function showPostUser(){
+  const postContainer = document.getElementById('allNewPost');
+        postContainer.innerHTML='';
+        db.collection("post").onSnapshot((querySnapshot) => {
+        querySnapshot.forEach((doc) => {;
+        //console.log(`${doc.id} => ${doc.data().post}`);
+        let postElement = document.createElement('div');
+        let postNew = `
+        <div class="containerPostPrint">
+        <div class="containerUser">
+          <img class="imgUser" src="img/user.svg"></img> <p class="nameUser">${doc.data().user}</p>
+          </div>
+          <div class="btnIcon">
+          <button type="submit" class="editBtn" ${doc.id}, ${doc.data().post} ><img src="img/edit.svg" /></button> 
+          <button type="submit" class="deleteBtn" data-id="${doc.id}"><img data-id="${doc.id}" src="img/delete.svg" /></button> 
+          </div>
+          <p class="PostPrint">${doc.data().post}</p>
+          <textarea class="answer" id="answer"> </textarea>
+          <br>
+          <div class="sectionLikes">
+          <p class="nroLikes">0<img class="imgLikes" src="img/like.svg" /></p>
+          </div>
+          <button class="btnAnswer" data-id="${doc.id}">Responder</button>
```

Comment:
Podrían darle formato a su código, esto facilita la lectura a cualquiera que lo lea ;) 
```suggestion
          <img class="imgUser" src="img/user.svg"></img> <p class="nameUser">${doc.data().user}</p>
          </div>
          <div class="btnIcon">
             <button type="submit" class="editBtn" ${doc.id}, ${doc.data().post} ><img src="img/edit.svg" /></button> 
             <button type="submit" class="deleteBtn" data-id="${doc.id}"><img data-id="${doc.id}" src="img/delete.svg" /></button> 
          </div>
          <p class="PostPrint">${doc.data().post}</p>
          <textarea class="answer" id="answer"> </textarea>
          <br>
          <div class="sectionLikes">
              <p class="nroLikes">0<img class="imgLikes" src="img/like.svg" /></p>
          </div>
          <button class="btnAnswer" data-id="${doc.id}">Responder</button>
```